packages/next-intl/src/middleware/middleware.tsx

```diff
@@ -183,6 +182,37 @@ export default function createMiddleware<Locales extends AllLocales>(
             )
           );
         }
+      } else {
+        const pathnameWithoutLocale =
+          pathname.split(`/${locale}`)[1] ?? pathname;
```

Comment:
Be careful with these matchings. This will be valid for example for the `/enterprise` page when locale is `en`.

May be would be better to do it like this:

```js
const pathnameWithoutLocale = pathname.split(new RegExp(`\/${lang}(?:\/|$)`))[1] ?? pathname;
```
or
```js
const matchedPath = pathname.match(new RegExp(`\/${lang}(?<path>\/.*|$)`));
const pathnameWithoutLocale = matchedPath?.groups.path ?? pathname
```
or
```js
const segments = pathname.split('/');
if (segments[1] === locale) segments.splice(1, 1);
const pathnameWithoutLocale = segments.join('/');
```

Reply:
@vordgi I don't understand, what is this `/enterprise` page? It's nowhere in tests nor exemples. This line of code is not about matching but rather removing locale from the url.

The matching is done below, by checking if the actual path exists in next intl paths mapping.

In any case, last suggestion is good !

Reply:
Just imagine that the locale is `en` and we got it from the domain (e.g. `en.example.com`), not from the path. The current path is `/enterprise` (`en.example.com/enterprise`).

Your logic (```pathname.split(`/${locale}`)[1] ?? pathname```) will remove `/en` from the path and keep `terprise` (`'/enterprise'.split('/en')[1] ?? '/enterprise'`)

Reply:
😮 I see now... thanks for explanation and suggestion!